generator/operation.go

```diff
@@ -1044,15 +913,208 @@ func (b *codeGenOpBuilder) MakeParameter(receiver string, resolver *typeResolver
+		Path:             res.Path,
+		Name:             b.Operation.ID + "ParamsBody",
+		Receiver:         res.ReceiverName,
+		ValueExpr:        res.ReceiverName,
+		IndexVar:         res.IndexVar,
+		Schema:           *sch,
+		Required:         false, // Required in body is managed independently from validations
+		TypeResolver:     rslv,
+		Named:            named,
+		IncludeModel:     true,
+		IncludeValidator: b.IncludeValidator,
+		ExtraSchemas:     make(map[string]GenSchema),
+	}
+	if err := sc.makeGenSchema(); err != nil {
+		return err
+	}
+	// TODO: lift nested extra schemas
+
+	schema := sc.GenSchema
+	if named {
+		if b.ExtraSchemas == nil {
+			b.ExtraSchemas = make(map[string]GenSchema)
+		}
+		b.ExtraSchemas[b.Operation.ID+"ParamsBody"] = schema
+	}
+	if schema.IsAnonymous {
+		// a generated name for anonymous parameter in body
+		nm := swag.ToGoName(b.Operation.ID + " Body")
+
+		hasProperties := len(schema.Properties) > 0
+		isInterface := schema.IsInterface
+		hasValidations := schema.HasValidations
+
+		// for complex anonymous objects, produce an extra schema
+		if hasProperties {
+			if b.ExtraSchemas == nil {
+				b.ExtraSchemas = make(map[string]GenSchema)
+			}
+			schema.Name = nm
+			schema.GoType = nm
+			schema.IsAnonymous = false
+			b.ExtraSchemas[nm] = schema
+
+			// constructs new schema to refer to the newly created type
+			schema = GenSchema{}
+			schema.IsAnonymous = false
+			schema.IsComplexObject = true
+			schema.SwaggerType = nm
+			schema.HasValidations = hasValidations
+			schema.GoType = nm
+		} else if isInterface {
+			schema = GenSchema{}
+			schema.IsAnonymous = false
+			schema.IsComplexObject = false
+			schema.IsInterface = true
+			schema.HasValidations = false
+			schema.GoType = iface
+		}
+	}
+
+	res.Schema = &schema
+	res.Schema.Required = res.Required
+
+	// build Child items for nested slices and maps
+	var items *GenItems
+	res.KeyVar = "k"
+	res.Schema.KeyVar = "k"
+	if schema.IsMap && !schema.IsInterface {
+		items = b.MakeBodyParameterItemsAndMaps(res, res.Schema.AdditionalProperties)
+	} else if schema.IsArray {
+		items = b.MakeBodyParameterItemsAndMaps(res, res.Schema.Items)
+	} else {
+		items = new(GenItems)
+	}
+
+	// templates assume at least one .Child != nil
+	res.Child = items
+	schema.HasValidations = schema.HasValidations || items.HasValidations
+
+	res.resolvedType = schema.resolvedType
+
+	// simple and schema views share the same validations
+	res.sharedValidations = schema.sharedValidations
+	res.ZeroValue = schema.Zero()
+	return nil
+}
+
+// MakeBodyParameterItemsAndMaps clones the .Items schema structure (resp. .AdditionalProperties) as a .GenItems structure
+// for compatibility with simple param templates.
+//
+// Constructed children assume simple structures: any complex object is assumed to be resolved by a model or extra schema definition
+func (b *codeGenOpBuilder) MakeBodyParameterItemsAndMaps(res *GenParameter, it *GenSchema) *GenItems {
```

Comment:
cyclomatic complexity 17 of func `(*codeGenOpBuilder).MakeBodyParameterItemsAndMaps` is high (> 10)